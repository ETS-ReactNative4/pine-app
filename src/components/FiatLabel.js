import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { getRegionLocale, DECIMAL_SEPARATOR } from '../localization';
import StyledText from './StyledText';

const formatAmount = (amount, currency) => {
  if (amount === null) {
    return `- ${currency}`;
  }

  const formatOptions = {
    style: 'currency',
    currencyDisplay: 'code',
    currency
  };

  let formattedAmount = new Intl.NumberFormat(getRegionLocale(), formatOptions).format(amount);

  // Remove the currency code - it will be added at the end (some locales will add it as a prefix).
  formattedAmount = formattedAmount.replace(currency, '').trim();

  // Remove decimals when they're all zeros, e.g. '.00'.
  formattedAmount = formattedAmount.replace(new RegExp(`\\${DECIMAL_SEPARATOR}[0]+$`), '');

  return `${formattedAmount} ${currency}`;
};

export default class FiatLabel extends Component {
  render() {
    const { amount, currency } = this.props;
    const label = formatAmount(amount, currency);

    return (
      <StyledText style={this.props.style}>
        {label}
      </StyledText>
    );
  }
}

FiatLabel.propTypes = {
  style: PropTypes.any,
  amount: PropTypes.number,
  currency: PropTypes.string.isRequired
};
