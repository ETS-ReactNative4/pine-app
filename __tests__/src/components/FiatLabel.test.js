import React from 'react';
import renderer from 'react-test-renderer';
import FiatLabel from '../../../src/components/FiatLabel';

describe('FiatLabel', () => {
  describe('when currency is USD', () => {
    it('renders 0 USD correctly', () => {
      const tree = renderer.create(
        <FiatLabel amount={0} currency='USD' />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('renders 2 USD correctly', () => {
      const tree = renderer.create(
        <FiatLabel amount={2} currency='USD' />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('renders 2.00 USD correctly', () => {
      const tree = renderer.create(
        <FiatLabel amount={2.00} currency='USD' />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('renders 2.30 USD correctly', () => {
      const tree = renderer.create(
        <FiatLabel amount={2.30} currency='USD' />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('renders 4000 USD correctly', () => {
      const tree = renderer.create(
        <FiatLabel amount={4000} currency='USD' />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });

  describe('when currency is SEK', () => {
    it('renders 0 SEK correctly', () => {
      const tree = renderer.create(
        <FiatLabel amount={0} currency='SEK' />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('renders 2 SEK correctly', () => {
      const tree = renderer.create(
        <FiatLabel amount={2} currency='SEK' />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('renders 2.00 SEK correctly', () => {
      const tree = renderer.create(
        <FiatLabel amount={2.00} currency='SEK' />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('renders 2.30 SEK correctly', () => {
      const tree = renderer.create(
        <FiatLabel amount={2.30} currency='SEK' />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('renders 4000 SEK correctly', () => {
      const tree = renderer.create(
        <FiatLabel amount={4000} currency='SEK' />
      ).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
