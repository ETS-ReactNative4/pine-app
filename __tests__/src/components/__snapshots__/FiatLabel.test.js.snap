// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiatLabel when currency is SEK renders 0 SEK correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  0 SEK
</Text>
`;

exports[`FiatLabel when currency is SEK renders 2 SEK correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  2 SEK
</Text>
`;

exports[`FiatLabel when currency is SEK renders 2.00 SEK correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  2 SEK
</Text>
`;

exports[`FiatLabel when currency is SEK renders 2.30 SEK correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  2.30 SEK
</Text>
`;

exports[`FiatLabel when currency is SEK renders 4000 SEK correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  4 000 SEK
</Text>
`;

exports[`FiatLabel when currency is USD renders 0 USD correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  0 USD
</Text>
`;

exports[`FiatLabel when currency is USD renders 2 USD correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  2 USD
</Text>
`;

exports[`FiatLabel when currency is USD renders 2.00 USD correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  2 USD
</Text>
`;

exports[`FiatLabel when currency is USD renders 2.30 USD correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  2.30 USD
</Text>
`;

exports[`FiatLabel when currency is USD renders 4000 USD correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000000",
        "fontFamily": "System",
        "fontWeight": "400",
      },
      undefined,
    ]
  }
>
  4 000 USD
</Text>
`;
